spi/transfer-spi/src/main/java/org/eclipse/dataspaceconnector/spi/types/domain/transfer/TransferProcess.java

```diff
@@ -419,6 +425,11 @@ public Builder type(Type type) {
             return this;
         }
 
+        public Builder createdTimestamp(long value) {
+            process.createdTimestamp = value;
```

Comment:
maybe the `build()` method could initialize the `createdTimestamp = Instant.now().epochMilli()` if it is null? Alternatively this could also be done in the private CTor of `TransferProcess`

Reply:
For defensive reasons I think it's better to do it this way to avoid side effects. For example, if the row is null in a SQL backend, the behavior you suggest would erroneously populate it with the current date.